puffin_egui/src/flamegraph.rs

```diff
@@ -782,60 +819,94 @@ fn paint_merge_scope(
             egui::show_tooltip_at_pointer(&info.ctx, Id::new("puffin_profiler_tooltip"), |ui| {
-                merge_scope_tooltip(ui, merge, info.num_frames);
+                merge_scope_tooltip(ui, info.scope_collection, merge, info.num_frames);
             });
         }
     }
 
-    Ok(result)
+    result
+}
+
+fn paint_scope_details(ui: &mut Ui, scope_id: ScopeId, data: &str, scope_details: &ScopeDetails) {
+    let scope_type = scope_details.scope_type();
+
+    egui::Grid::new("merge_scope_tooltip")
+        .num_columns(2)
+        .show(ui, |ui| {
+            ui.monospace("id");
+            ui.monospace(format!("{}", scope_id.0));
+            ui.end_row();
+
+            ui.monospace("name");
+            ui.monospace(format!("{}", scope_type.name()));
+            ui.end_row();
+
+            if !scope_details.file_path.is_empty() {
+                ui.monospace("location");
+                ui.monospace(scope_details.location());
+                ui.end_row();
+            }
+
+            if !data.is_empty() {
+                ui.monospace("data");
+                ui.monospace(data.as_str());
+                ui.end_row();
+            }
+
+            ui.monospace("type");
+            ui.monospace(scope_type.type_str());
+            ui.end_row();
+        });
 }
 
-fn merge_scope_tooltip(ui: &mut egui::Ui, merge: &MergeScope<'_>, num_frames: usize) {
+fn merge_scope_tooltip(
+    ui: &mut egui::Ui,
+    scope_collection: &ScopeCollection,
+    merge: &MergeScope<'_>,
+    num_frames: usize,
+) {
     #![allow(clippy::collapsible_else_if)]
 
-    ui.monospace(format!("id:       {}", merge.id));
-    if !merge.location.is_empty() {
-        ui.monospace(format!("location: {}", merge.location));
-    }
-    if !merge.data.is_empty() {
-        ui.monospace(format!("data:     {}", merge.data));
-    }
-    ui.add_space(8.0);
+    let Some(scope_details) = scope_collection.fetch_by_id(&merge.id) else {
+        return;
+    };
+
+    paint_scope_details(ui, merge.id, &merge.data, scope_details);
+
+    ui.separator();
 
     if num_frames <= 1 {
         if merge.num_pieces <= 1 {
             ui.monospace(format!(
-                "duration: {:7.3} ms",
+                "duration:\t{:.3} ms",
                 to_ms(merge.duration_per_frame_ns)
             ));
         } else {
             ui.monospace(format!("sum of {} scopes", merge.num_pieces));
             ui.monospace(format!(
-                "total: {:7.3} ms",
+                "total:\t{:.3} ms",
                 to_ms(merge.duration_per_frame_ns)
             ));
             ui.monospace(format!(
-                "mean:  {:7.3} ms",
+                "mean:\t{:.3} ms",
                 to_ms(merge.duration_per_frame_ns) / (merge.num_pieces as f64),
             ));
-            ui.monospace(format!("max:   {:7.3} ms", to_ms(merge.max_duration_ns)));
+            ui.monospace(format!("max:\t{:.3} ms", to_ms(merge.max_duration_ns)));
```

Comment:
Are you sure `\t` aligns subsequent labels? Have you tested the UI?

Reply:
see printscreens above, does that look correct to you?

Reply:
I don't see any screenshots with total/mean/max, so I made one:

![image](https://github.com/EmbarkStudios/puffin/assets/1148717/39072741-82bb-4f8c-9744-01659243f647)

and so no, it doesn't

Reply:
the current format is a bit confusing to me - manual spaces with spaces specified by {}. Had a couple of issues with it but I'll try revert back to that